components/UI/profileCard/profileCard.tsx

```diff
@@ -0,0 +1,180 @@
+import { hexToDecimal } from "@utils/feltService";
+import { calculatePercentile } from "@utils/numberService";
+
+const ProfileCard: FunctionComponent<ProfileCard> = ({
+  rankingData,
+  identity,
+  leaderboardData,
+  addressOrDomain,
+  isOwner,
+}) => {
+  const [copied, setCopied] = useState(false);
+  const sinceDate = useCreationDate(identity);
+  const { data: profileData } = useStarkProfile({ address: identity.owner });
+  const [showSharePopup, setShowSharePopup] = useState(false);
+  const [userPercentile, setUserPercentile] = useState("");
+  const [userXp, setUserXp] = useState<number>();
+
+  const copyToClipboard = () => {
+    setCopied(true);
+    navigator.clipboard.writeText(identity?.owner as string);
+    setTimeout(() => {
+      setCopied(false);
+    }, 1500);
+  };
+
+  const computeData = useCallback(() => {
+    if (rankingData && identity.owner && leaderboardData) {
+      const user = rankingData.ranking.find(
+        (user) => user.address === hexToDecimal(identity.owner)
+      );
+      if (user) {
+        if (!leaderboardData.position) {
+          setUserPercentile("NA");
+          return;
+        }
+        const computedUserPercentile = calculatePercentile(
+          leaderboardData?.position ?? 0,
+          leaderboardData?.total_users ?? 0
+        ).toString();
+        setUserPercentile(computedUserPercentile);
+        setUserXp(user.xp);
+      }
+    }
+  }, [rankingData, identity, leaderboardData]);
+
+  useEffect(() => {
+    computeData();
+  }, [rankingData, identity, leaderboardData]);
+
+  return (
+    <>
+      <div className={styles.dashboard_profile_card}>
+        <div className={`${styles.left} ${styles.child}`}>
+          <div className={styles.profile_picture_div}>
+            {profileData?.profilePicture ? (
+              <img src={profileData?.profilePicture} className="rounded-full" />
+            ) : (
+              <ProfilIcon width={"120"} color={theme.palette.secondary.main} />
+            )}
+          </div>
+        </div>
+        <div className={`${styles.center} ${styles.child}`}>
+          <p className={styles.accountCreationDate}>
+            {sinceDate ? `${sinceDate}` : ""}
+          </p>
+          <h2 className={styles.profile_name}>{identity.domain.domain}</h2>
+          <div className={styles.address_div}>
+            <div onClick={() => copyToClipboard()}>
+              {!copied ? (
+                <Tooltip title="Copy" arrow>
+                  <div onClick={() => copyToClipboard()}>
+                    <CopyIcon width={"20"} color="#F4FAFF" />
+                  </div>
+                </Tooltip>
+              ) : (
+                <VerifiedIcon width={"20"} />
+              )}
+            </div>
+            <p className={styles.addressText}>
+              {typeof addressOrDomain === "string" &&
+                minifyAddressFromStrings(
+                  [addressOrDomain, identity?.owner || ""],
+                  8
+                )}
```

Comment:
You need to make this snippet easier. 

I don't understand why you give an array when you could actually make sure you're giving an address to the function. 

Why can `addressOrDomain` be string[] ?

Reply:
fixed. This was done in the lands version earlier so i used the same implementation. I fix it now with a improved version